fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/jbatch/bulkdata/export/fast/ResourcePayloadReader.java

```diff
@@ -0,0 +1,683 @@
+     */
+    @Inject
+    @BatchProperty(name = Constants.COS_SRVINST_ID)
+    String cosSrvinstId;
+
+    /**
+     * The Cos End point URL.
+     */
+    @Inject
+    @BatchProperty(name = Constants.COS_ENDPOINT_URL)
+    String cosEndpointUrl;
+
+    /**
+     * The Cos End point location.
+     */
+    @Inject
+    @BatchProperty(name = Constants.COS_LOCATION)
+    String cosLocation;
+
+    /**
+     * The Cos bucket name.
+     */
+    @Inject
+    @BatchProperty(name = Constants.COS_BUCKET_NAME)
+    String cosBucketName;
+
+    /**
+     * The Cos bucket path prefix.
+     */
+    @Inject
+    @BatchProperty(name = Constants.EXPORT_COS_OBJECT_PATHPREFIX)
+    String cosBucketPathPrefix;
+
+    /**
+     * If use IBM credential or Amazon secret keys.
+     */
+    @Inject
+    @BatchProperty(name = Constants.COS_IS_IBM_CREDENTIAL)
+    String cosCredentialIbm;
+
+    /**
+     * Fhir export format.
+     */
+    @Inject
+    @BatchProperty(name = Constants.EXPORT_FHIR_FORMAT)
+    protected String fhirExportFormat;
+
+    /**
+     * Fhir Search from date.
+     */
+    @Inject
+    @BatchProperty(name = Constants.EXPORT_FHIR_SEARCH_FROMDATE)
+    String fhirSearchFromDate;
+
+    /**
+     * Fhir search to date.
+     */
+    @Inject
+    @BatchProperty(name = Constants.EXPORT_FHIR_SEARCH_TODATE)
+    String fhirSearchToDate;
+
+    /**
+     * Fhir export type filters. Ignored for the basicsystem export
+     */
+    @Inject
+    @BatchProperty(name = Constants.EXPORT_FHIR_SEARCH_TYPEFILTERS)
+    String fhirTypeFilters;
+
+    /**
+     * Fhir search page size. Not used
+     */
+    @Inject
+    @BatchProperty(name = Constants.EXPORT_FHIR_SEARCH_PAGESIZE)
+    String fhirSearchPageSize;
+
+    @Inject
+    @BatchProperty(name = Constants.INCOMING_URL)
+    String incomingUrl;
+
+    // Control the number of records to read in each "item".
+    @Inject
+    @BatchProperty(name = Constants.COS_BUCKET_FILE_MAX_RESOURCES)
+    String cosBucketFileMaxResources;
+    int resourcesPerItem = Constants.DEFAULT_MAX_RESOURCES_PER_ITEM;
+
+    // The Java Batch context object
+    @Inject
+    StepContext stepCtx;
+
+    // The context object representing this export job
+    @Inject
+    JobContext jobContext;
+
+    // Track the resources we've processed sharing the most recent timestamp - for correct checkpoint
+    private Set<Long> resourcesForLastTimestamp = new HashSet<>();
+
+    // Each time the timestamp changes, we need to clear resourcesForLastTimestamp
+    private Instant lastTimestamp;
+
+    // The last_modified timestamp to start scanning from
+    private Instant fromLastModified;
+
+    // An end time to scan to, if any
+    private Instant toLastModified;
+
+    // Cap the part upload size to avoid local memory issues. Also need to avoid transaction timeout
+    private long partUploadTriggerSize = Constants.COS_PART_MINIMALSIZE * 10L;
+
+    // How large should a single COS item (file) be
```

Comment:
is "COS item" here synonymous with "COS object"?
is it synonymous with "Java batch item"?

Reply:
Yeah agree about consistency. I wanted to move away from "file" - not sure how I landed on "item" but we can change to object if needs be.